docs/guides/custom-plugins.md

```diff
@@ -0,0 +1,229 @@
+# Creating custom pyscript plugins
```

Comment:
This writeup is really well done, and is an excellent example of how to use Python to create customElements. But I think it would be nice to have an example that uses the plugin-based lifecycle events, seeing as `register_custom_element` doesn't really use them either.

What I mean is - you can obtain the same functionality as the given example, in a plain-old pyscript tag:

```html
<py-script>
from pyscript import Plugin

@Plugin.register_custom_element(None, "py-hello-world")
class PyHelloWorld:
    def __init__(self, element):
        self.element = element
    def connect(self):
        self.element.innerHTML = "<h1>Hello World!</h1>"
</py-script>
```

Note the `None` in the decorator call (in place of `self`) - `register_custom_element` doesn't actually need to be associated with a specific plugin at all, it's just oddly attached to the Plugins API...

This is really a quibble with the create-custom-element API, but perhaps it would be nice to have different examples for plugins (using the callbacks) and `register_custom_element`.

Reply:
That's a great point, perhaps this should be moved to the tutorial section and we should create a proper guide using the lifecycle since that seems a bit more in-depth article - also this guide became just a small tutorial haha